### PR TITLE
MON-3701: clean-up injection of CA bundle for user Prometheus

### PR DESCRIPTION
**Pull request by simonpasquier**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: assets/prometheus-user-workload/prometheus.yaml [hunk 1]
```diff
@@ -262,3 +265,10 @@ spec:
         memory: 100Mi
     version: 0.35.0
   version: 2.52.0
+  volumes:
+  - configMap:
+      items:
+      - key: ca-bundle.crt
```
Comment (machine424) [inline assets/prometheus-user-workload/prometheus.yaml:271]: I don't know if kube makes sure that the key value isn't empty before starting the Pod and if the value can be empty during a transient/temporary period...
(but even in that case, maybe, the Pod will converge)

File: pkg/manifests/manifests_test.go [hunk 2]
```diff
@@ -1765,6 +1763,19 @@ func TestPrometheusUserWorkloadConfiguration(t *testing.T) {
 	if !reflect.DeepEqual(*f.config.UserWorkloadConfiguration.Prometheus.Resources, p.Spec.Resources) {
 		t.Fatal("Prometheus UWM resources not configured correctly")
 	}
+
+	for _, container := range p.Spec.Containers {
+		switch container.Name {
```
Comment (machine424) [inline pkg/manifests/manifests_test.go:1768]: nit: maybe we can raise in `default:` so we have a safe guard.

**Review by machine424 — COMMENTED** [review]
Thanks.

File: pkg/tasks/prometheus_user_workload.go [hunk 3]
```diff
@@ -318,6 +312,12 @@ func (t *PrometheusUserWorkloadTask) create(ctx context.Context) error {
 		}
 	}
 
+	// TODO(simonpasquier): remove this step after OCP 4.17 is released.
```
Comment (machine424) [inline pkg/tasks/prometheus_user_workload.go:315]: I created https://issues.redhat.com/browse/MON-3880 for those

File: assets/prometheus-user-workload/prometheus.yaml (hunk 1 above)
Comment (simonpasquier) [reply assets/prometheus-user-workload/prometheus.yaml:271]: no Kubernetes wouldn't check that but yes it should converge eventually.

File: pkg/manifests/manifests.go [hunk 4]
```diff
@@ -1751,13 +1741,7 @@ func (f *Factory) PrometheusUserWorkload(grpcTLS *v1.Secret, trustedCABundleCM *
 				PeriodSeconds:    15,
 				FailureThreshold: 240,
 			}
-			// Support CA bundles for Prometheus UWM.
-			if trustedCABundleVolumeName != "" {
-				p.Spec.Containers[i].VolumeMounts = append(
-					p.Spec.Containers[i].VolumeMounts,
-					trustedCABundleVolumeMount(trustedCABundleVolumeName),
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/manifests/manifests.go:1758]: No it's used for telemeter

File: pkg/manifests/manifests_test.go (hunk 2 above)
Comment (simonpasquier) [reply pkg/manifests/manifests_test.go]: changed a bit the test to make it more explicit indeed.

**Pull request merged**